cgi-bin/DW/Controller/Manage/Circle/Edit.pm

```diff
@@ -0,0 +1,313 @@
+#!/usr/bin/perl
+#
+# DW::Controller::Manage::Circle::PopSubscriptions
+#
+# Page that shows a sorted list of popular accounts in the user's circle.
+# User can pick which circle group to base these calculations on by selecting
+# from a drop-down menu. Results are displayed in three sections: personal
+# accounts, community accounts, feed accounts. Page for listing subscriptions
+# that are popular with other members of the user's circle.
+#
+# Authors:
+#   Rebecca Freiburg <beckyvi@gmail.com>   (BML version)
+#   Jen Griffin <kareila@livejournal.com>  (TT conversion)
+#
+# Copyright (c) 2009-2022 by Dreamwidth Studios, LLC.
+#
+# This program is free software; you may redistribute it and/or modify it under
+# the same terms as Perl itself. For a copy of the license, please reference
+# 'perldoc perlartistic' or 'perldoc perlgpl'.
+#
+
+package DW::Controller::Manage::Circle::Edit;
+
+use strict;
+
+use DW::Routing;
+use DW::Template;
+use DW::Controller;
+use LJ::JSON;
+
+my $popsub_path = '/manage/circle/edit';
+
+DW::Routing->register_string( '/manage/circle/edit', \&edit_handler, app => 1 );
+
+sub edit_handler {
+    my ( $ok, $rv ) = controller();
+    return $rv unless $ok;
+
+    my $r = DW::Request->get;
+
+    my $remote = $rv->{remote};
+
+    my $GET  = $r->get_args;
+    my $POST = $r->post_args;
```

Comment:
It is not safe to use post_args (or take actions, like below) without checking form_auth (and I can't find a call to check_form_auth.)

anyway, controller does not do this automatically, but you have to tell it to with form_auth => 1.